thor/src/protocols/splice.rs

```diff
@@ -100,20 +156,58 @@ impl State0 {
         })
         .expect("Only fails if script_sig or witness is empty which is not the case.");
 
+        let mut our_balance = self.previous_balance.ours;
+        let mut their_balance = self.previous_balance.theirs;
+
+        let mut splice_in_inputs = vec![];
+
+        if let Some(splice_in) = self.splice_in_self.clone() {
+            splice_in_inputs.push(splice_in.input_psbt);
+            our_balance += splice_in.amount;
+        }
+
+        if let Some(splice_in) = splice_in_other {
+            splice_in_inputs.push(splice_in.input_psbt);
+            their_balance += splice_in.amount;
+        }
+
+        // Sort the PSBT inputs based on the ascending lexicographical order of
+        // bytes of their consensus serialization. Both parties _must_ do this so that
+        // they compute the same splice transaction.
+        splice_in_inputs.sort_by(|a, b| {
+            serialize(a)
+                .partial_cmp(&serialize(b))
+                .expect("comparison is possible")
+        });
+
+        // The previous funding psbt MUST be the first input
+        let mut inputs = vec![previous_funding_psbt];
+
+        inputs.append(&mut splice_in_inputs);
+
         let balance = Balance {
-            ours: self.previous_balance.ours - Amount::from_sat(crate::TX_FEE / 2),
-            theirs: self.previous_balance.theirs - Amount::from_sat(crate::TX_FEE / 2),
+            ours: our_balance,
+            theirs: their_balance,
         };
 
-        let TX_f = FundingTransaction::new(vec![previous_funding_psbt], [
+        let TX_f = SpliceTransaction::new(inputs, [
             (self.x_self.public(), balance.ours),
             (self.X_other.clone(), balance.theirs),
         ])?;
 
+        // TODO: Clean-up the signature/PSBT mix (if possible)
+
+        // Signed to spend TX_f
         let sig_TX_f = TX_f.sign_once(self.x_self.clone(), &self.previous_TX_f);
 
+        // Signed to spend the splice-in input
+        let splice_in_signed_TX_f_self = match self.splice_in_self {
+            Some(_) => Some(wallet.sign_funding_psbt(TX_f.clone().into_psbt()?).await?),
+            None => None,
+        };
```

Comment:
🚧
We should not sign our `splice_in` tx yet but only once all `TX_c` and `TX_s` are completed, i.e. in the last step in `State3`. It might be a security issue to keep it in memory and it is not used until sent to the other party in `Step3`.